src/nodeClient.js

```diff
@@ -1,43 +1,24 @@
-const Web3 = require('web3');
+const ethers = require('ethers');
 const { helpers } = require('leap-core');
 
 const { formatHostname, advanceBlocks, sleep } = require('./helpers');
 
 let idCounter = 0;
 
-class Node {
+class Node extends helpers.LeapEthers {
   constructor(hostname, jsonrpcPort) {
-    this.id = idCounter++;
-    this.web3 = helpers.extendWeb3(new Web3(formatHostname(hostname, jsonrpcPort)));
-  }
+    const provider = new ethers.providers.JsonRpcProvider(formatHostname(hostname, jsonrpcPort))
+    super(provider);
 
-  async send(method, params) {
-    return await new Promise(
-      (resolve, reject) => {
-        this.web3.currentProvider.send(
-          { jsonrpc: '2.0', id: 42, method: method, 'params': params },
-          (err, res) => { if (err) { return reject(err); } resolve(res); }
-        );
-      }
-    );
+    this.id = idCounter++;
   }
 
   async sendTx(tx) {
     // workaround different transaction hashes (leap-core bug)
     let txHash;
 
     try {
-      // web3 hangs here on invalid txs, trying to get receipt?
-      // await this.web3.eth.sendSignedTransaction(tx.hex());
-      const resp = await new Promise(
-        (resolve, reject) => {
-          this.web3.currentProvider.send(
-            { jsonrpc: '2.0', id: 42, method: 'eth_sendRawTransaction', 'params': [tx.hex()] },
-            (err, res) => { if (err) { return reject(err); } resolve(res); }
-          );
-        }
-      );
-      txHash = resp.result;
+      txHash = await this.provider.send('eth_sendRawTransaction', [tx.hex()]);
```

Comment:
can be replaced with https://github.com/leapdao/leap-core/blob/master/lib/helpers.js#L343